### DISCUSSION TITLE
repair_design hanging

### DISCUSSION THREAD

**Discussion by msingh9** [The-OpenROAD-Project/OpenROAD#3073] [Q&A]
I have "repair_design" to repair slew and cap, but this has been running for long time with no output. I think it is hanging. My design is not that big. I think it will be useful to get some sense of how much work has been completed, like work progress from these functions. Maybe I should file this a enhancement request? Is there a way to print some message while it is working?

**Comment by maliberty** [comment]
You could file it as an enhancement request or try to tackle it yourself and file a PR.

**Comment by vijayank88** [comment]
@msingh9 
Are you using standalone OpenROAD or with OpenROAD-flow-scripts?

**Reply by msingh9** [reply]
I am using stand alone openroad. It works some-times. I have this general comment that if we can include a progress bar or something like that so we know that it is working and making progress. My next issue is with this command "detailed_placement -max_displacement 50". Now it is sitting for about an hour. I don't know if this is still making progress or hung. 

I am not very knowledgeable in software development, so if I have to tackle this by myself, it may take a lot longer :). In long term, I will like to make changes but as of now I need help.

**Comment by maliberty** [comment]
How large is your design?

**Reply by msingh9** [reply]
It is about 240K instances. I am not sure what I did, but it doesn't hang any more. So, I will close this discussion. I like to contribute to the good and awesome work you all have put in, once I am ready for it. I would like to add some kind of progress bar to all methods, so we get an idea of the progress as well as if something is hung. It is not an easy task so I don't yet know how to do it in here, but I will learn in long term.
